esmvalcore/preprocessor/_time.py

```diff
@@ -725,3 +755,81 @@ def timeseries_filter(cube, window, span,
                                weights=wgts)
 
     return cube
+
+
+def resample_hours(cube, interval, offset=0):
+    """Convert x-hourly data to y-hourly by eliminating extra timesteps.
+
+    Convert x-hourly data to y-hourly (y > x) by eliminating the extra
+    timesteps. This is intended to be used only with instantaneous values.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Input cube.
+    interval: int
+        The period (hours) of the desired data.
+    offset: int, optional
+        The firs hour (hours) of the desired data.
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube with the new frequency.
+
+    Raises
+    ------
+    ValueError:
+        The specified frequency is not a divisor of 24.
+    """
+    allowed_intervals = (1, 2, 3, 4, 6, 12)
+    if interval not in allowed_intervals:
+        raise ValueError(
+            f'The number of hours must be one of {allowed_intervals}')
+    if offset >= interval:
+        raise ValueError(f'The offset ({offset}) must be lower than '
+                         f'the interval ({interval})')
+    time = cube.coord('time')
+    cube_period = time.cell(1).point - time.cell(0).point
+    if cube_period.total_seconds() / 3600 >= interval:
+        raise ValueError(f"Data period ({cube_period}) should be lower than "
+                         f"the interval ({interval})")
+    hours = range(0 + offset, 24, interval)
+    select_hours = iris.Constraint(time=lambda cell: cell.point.hour in hours)
+    return cube.extract(select_hours)
+
+
+def resample_time(cube, month=None, day=None, hour=None):
+    """Change frequency of data by resampling it.
+
+    Converts data from one frequency to another by extracting the timesteps
+    that match the provided month, day and/or hour. This is meant to be used
+    with instantaneous values when computing statistics is not desired.
```

Comment:
I added the example from above, but maybe it is worth it to some more examples.

```suggestion
    with instantaneous values when computing statistics is not desired.
    
    For example, `resample_time(cube, hour=6)` -> Daily values taken at 6:00.
```